external-crates/move/crates/move-compiler/tests/move_check/parser/cast_type_with_type_args.exp

```diff
@@ -0,0 +1,9 @@
+error[E01002]: unexpected token
+  ┌─ tests/move_check/parser/cast_type_with_type_args.move:4:22
+  │
+4 │         (0 as Cup<u8>);
+  │                      ^
+  │                      │
+  │                      Unexpected ')'
+  │                      Expected an expression term
```

Comment:
This error is amazingly bad

Reply:
Oof -- maybe the resilience comment above will help?

Reply:
I don't think so? The issue is its treating the first `<` as a BinOp, rather than the start of a type argument. So Its parsing `(Cup < u8) >` and is then reasonably surprised when it hits a `)`

I could make this whitespace sensitive?